app/assets/javascripts/discourse/tests/acceptance/search-test.js

```diff
@@ -131,58 +132,57 @@ acceptance("Search - Anonymous", function (needs) {
   });
 
   test("search scope", async function (assert) {
-    const firstResult =
-      ".search-menu .results .search-menu-assistant-item:first-child";
+    const contextSelector = ".search-menu .results .search-menu-assistant-item";
 
     await visit("/tag/important");
     await click("#search-button");
 
     assert.strictEqual(
-      query(firstResult).textContent.trim(),
+      queryAll(contextSelector)[0].firstChild.textContent.trim(),
```

Comment:
Instead of `[0].firstChild`/`[1].firstChild` you can do:

```suggestion
      query(".search-menu-assistant-item:nth-child(1)").textContent.trim(),
```

or even replace the whole assertion with:

```js
assert.dom(".search-menu-assistant-item:nth-child(1)").hasText(
  `${I18n.t("search.in")} test`,
  "contextual tag search is first available option with no term"
)
```

Reply:
This works for consistently getting the first item. I started with it and then spent a lot of time confused why matching the second item wasn't as consistent. It turns out  for some results,  there is a nested `ul` situation where the second child isn't a `search-menu-assistant-item`

<img width="506" alt="Screenshot 2023-03-08 at 8 57 36 AM" src="https://user-images.githubusercontent.com/849886/223672346-853fd757-570f-461e-ab7a-6f5ebefc8cd0.png">

Reply:
uhhh, ul in a ul… 😑 okay, let's go with this for now, and I'll look into fixing this

Reply:
oh, actually, I'm sure @janzenisaac will take care of that in #20482 😁